grouper/background/background_processor.py

```diff
@@ -99,28 +99,32 @@ def promote_nonauditors(self, session):
         # map from user object to names of audited groups in which
         # user is a nonauditor approver
         nonauditor_approver_to_groups = defaultdict(set)  # type: Dict[User, Set[str]]
-        # TODO(tyleromeara): replace with graph call
-        for group in get_audited_groups(session):
-            members = group.my_members()
-            # Go through every member of the group and add them to
-            # auditors group if they are an approver but not an
-            # auditor
-            for (type_, member), edge in members.iteritems():
-                # Auditing is already inherited, so we don't need to handle that here
-                if type_ == "Group":
+        user_is_auditor = {}  # type: Dict[str, bool]
+        for group_tuple in graph.get_groups(audited=True, directly_audited=False):
+            group_md = graph.get_group_details(group_tuple.groupname, expose_aliases=False)
+            # members = group.my_members()
+            # # Go through every member of the group and add them to
+            # # auditors group if they are an approver but not an
+            # # auditor
+            for username, user_md in group_md['users'].items():
+                if username not in user_is_auditor:
+                    user_perms = graph.get_user_details(username)['permissions']
+                    user_is_auditor[username] = any(
+                        [p['permission'] == PERMISSION_AUDITOR for p in user_perms])
+                if user_is_auditor[username]:
+                    # user is already auditor so can skip
                     continue
-                member = User.get(session, name=member)
-                member_is_approver = user_role_index(member, members) in APPROVER_ROLE_INDICES
-                member_is_auditor = user_has_permission(session, member, PERMISSION_AUDITOR)
```

Comment:
Can we not still use this function? Would simplify code, and you end up calling User.get later on anyway so doesn't hurt to pull that call up here.